### ISSUE TITLE
Rails + Docker Compose: Setting the hostname/IP of the tracer did not work as expected

### ISSUE THREAD

**Issue by iox** [DataDog/dd-trace-rb#682]
I'm running a Rails application using docker compose. I have added datadog as a linked docker service.

Then, I have tried to set the tracer to use the hostname of the datadog agent:

```ruby

Datadog.configure do |c|
  c.use :rails, service_name: 'medas-rails'
  c.use :sidekiq
  c.tracer hostname: 'datadog-agent'
end

```

But it didn't work, I could see in the logs that it was still trying to connect to localhost:8126.

So, after checking the source (https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/transport.rb#L54), I have added the following environment variable which solves the issue:

```
   - DD_AGENT_HOST=datadog-agent
```

I'm sharing this workaround here in case anyone encounters the same issue. Maybe I was using the wrong syntax in my Datadog configuration :).

**Comment by delner** [comment]
Well that's very curious if the `hostname` isn't working; the `c.tracer hostname: 'datadog-agent'` line should reconfigure the tracer and writer with the specified hostname via https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/tracer.rb#L149

Thanks for sharing this @iox!

**Comment by delner** [comment]
I tried to replicate this locally without success. I removed all Datadog env variables from the Docker container and configured the tracer with `c.tracer hostname: 'ddagent'` and it sent the traces successfully to my designated hostname.

I'd like to see a code sample that can replicate this. In the interim I'm going to close this issue as it can't be reproduced and there's a prescribed workaround.

Thanks for the report @iox.
